### PR TITLE
Feature/35 shared UI components tests

### PR DESCRIPTION
**Pull request by aashapovalov**
Screenshot 2025-09-18 at 13 58 36

## Summary

Add tests for shared UI components

## Changes

- **Dropdown**: add RTL tests for open/close, selection, and fallback label.  
- **JsonViewer**: add tests for text mode, line numbers, and scroll sync.  
- **KeyValueEditor**: add tests for dispatching trailing-empty logic on mount/update.  
- **KeyValueRow**: add tests for enabling rows, updates, and remove action.  
- **SignOutButton**: add tests for logout flow and navigation.  
- Fix `SignOutButton` to be a named export and update `Header` accordingly.  
- Install `@testing-library/user-event`.

### CODE REVIEW

File: __tests__/shared/ui/key-value-row-test.tsx
Comment (dariadrozdova) [inline __tests__/shared/ui/key-value-row-test.tsx:1]: `__tests__/shared/ui/key-value-row.test.tsx`

I suppose there is a typo in file name

**Review by dariadrozdova — CHANGES_REQUESTED** [review]
Great work!

The only thing I think is that all comments should be removed to avoid problems during cross-check.

File: __tests__/shared/ui/key-value-editor.test.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import React from "react";
+
+import { createAction } from "@reduxjs/toolkit";
+import { render } from "@testing-library/react";
+import { describe, expect, it, vi } from "vitest";
+
+import { KeyValueEditor } from "@shared/ui";
+
+// capture dispatched actions
```
Comment (dariadrozdova) [inline __tests__/shared/ui/key-value-editor.test.tsx]: please remove all comments from this file

File: __tests__/shared/ui/sign-out-button.test.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
+// __tests__/shared/ui/sign-out-button.test.tsx
```
Comment (dariadrozdova) [inline __tests__/shared/ui/sign-out-button.test.tsx]: please remove all comments from this file

File: __tests__/shared/ui/key-value-row-test.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
+import React from "react";
+
+import { createAction } from "@reduxjs/toolkit";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { beforeEach, describe, expect, it, vi } from "vitest";
+
+import { KeyValueRow } from "@/shared/ui";
+
+// dispatch spy
```
Comment (dariadrozdova) [inline __tests__/shared/ui/key-value-row-test.tsx]: please remove all comments from this file

**Review by dariadrozdova — APPROVED** [review]
